### PR TITLE
Issue #144 Add codefix for S1905 (redundant casts)

### LINKED ISSUE #144: Add codefix for S1905 (redundant casts)

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #144

### CODE REVIEW

File: src/Tests/SonarLint.UnitTest/Rules/RedundantCastTest.cs [hunk 1]
```diff
@@ -32,5 +32,16 @@ public void RedundantCast()
         {
             Verifier.VerifyAnalyzer(@"TestCases\RedundantCast.cs", new RedundantCast());
         }
+
+        [TestMethod]
+        [TestCategory("CodeFix")]
+        public void RedundantCast_CodeFix()
```
Comment (tvsonar) [inline src/Tests/SonarLint.UnitTest/Rules/RedundantCastTest.cs]: @dbolkensteyn This one fails at the moment, it needs the modification from #184

File: src/SonarLint/Rules/RedundantCastCodeFixProvider.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        }
+
+        private static async Task<Document> RemoveStaticMemberCall(Document document, MemberAccessExpressionSyntax memberAccess,
+            CancellationToken cancellationToken)
+        {
+            var invocation = memberAccess.Parent as InvocationExpressionSyntax;
+            if (invocation == null ||
+                !invocation.ArgumentList.Arguments.Any())
+            {
+                return document;
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/RedundantCastCodeFixProvider.cs]: @tvsonar it seems to me like this check should be done before the code fix is suggested. It's misleading to offer a codefix, and to later not fix anything in the end

File: src/SonarLint/Rules/RedundantCastCodeFixProvider.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private static async Task<Document> RemoveExtensionMethodCall(Document document, InvocationExpressionSyntax invocation,
+            CancellationToken cancellationToken)
+        {
+            var memberAccess = invocation.Expression as MemberAccessExpressionSyntax;
+            if (memberAccess == null)
+            {
+                return document;
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/RedundantCastCodeFixProvider.cs]: @tvsonar same thing here

File: src/SonarLint/Rules/RedundantCastCodeFixProvider.cs (hunk 2 above)
Comment (dbolkensteyn) [inline src/SonarLint/Rules/RedundantCastCodeFixProvider.cs]: or perhaps this actually is deadcode - in which case this hypothesis is better replaced by a `Deubg.Assert()` or something that will help to fix the code

Comment (dbolkensteyn) [inline src/SonarLint/Rules/RedundantCastCodeFixProvider.cs]: it actually seems like deadcode - the analyzer won't report if this hypothesis isn't true

**Comment by dbolkensteyn** [comment]
LGTM, but `x.OfType<int>().Cast<int>().ToList();` should be solvable all at once

**Comment by dbolkensteyn** [comment]
Let's got ahead with the current code fix - I've created another issue to improve it later on https://github.com/SonarSource/sonarlint-vs/issues/208

File: src/SonarLint/Rules/RedundantCastCodeFixProvider.cs (hunk 2 above)
Comment (tvsonar) [inline src/SonarLint/Rules/RedundantCastCodeFixProvider.cs]: @dbolkensteyn As you pointed out it is dead code, so I simply removed it.

**Pull request merged**